### PR TITLE
[Transaction][Buffer] handle command `EndTxnOnPartitiion`

### PR DESCRIPTION
**Pull request by zymap**
### Motivation

*Add handler for the command `EndTxnOnPartitioin*

### CODE REVIEW

**Comment by zymap** [comment]
@sijie Please help me take a look when you have time. Thanks

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java [hunk 1]
```diff
@@ -1358,6 +1360,57 @@ protected void handleGetSchema(CommandGetSchema commandGetSchema) {
...
+                                            ctx.writeAndFlush(
+                                                Commands.newEndTxnOnPartitionResponse(requestId, ServerError.UnknownError, err.getMessage()));
+                                            return null;
+                                        });
+                    } else if (commandEndTxnOnPartition.getTxnAction().equals(PulsarApi.TxnAction.ABORT)) {
+                        markerController.publishAbortMarker()
+                                        .thenCompose(ignore -> partition.getTxnBuffer(false))
+                                        .thenCompose(buffer -> buffer.abortTxn(txnID))
+                                        .exceptionally(err -> {
+                                            log.error("Abort txn error : ", err);
```
Comment (sijie) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java]: same comment as above

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java [hunk 2]
```diff
@@ -1533,4 +1586,49 @@ public boolean isBatchMessageCompatibleVersion() {
     public String getClientVersion() {
         return clientVersion;
     }
+
+    /**
+     * The transaction marker controller controls the commit marker or the abort marker publish to the specified topic.
+     * If it is a commit marker, we need to wait for the position it committed at the topic.
+     */
+    final static class TxnMarkerController implements Topic.PublishContext {
```
[the remark this answers is not in the document]
Comment (zymap) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java]: I need. I will fix it.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java [hunk 3]
```diff
@@ -1358,6 +1360,57 @@ protected void handleGetSchema(CommandGetSchema commandGetSchema) {
...
+                                                        ctx.writeAndFlush(
+                                                            Commands.newEndTxnOnPartitionResponse(requestId, txnID.getLeastSigBits(), txnID.getMostSigBits())))
+                                        .exceptionally(err -> {
+                                            log.error("Commit txn error :", err);
+                                            ctx.writeAndFlush(
+                                                Commands.newEndTxnOnPartitionResponse(requestId, ServerError.UnknownError, err.getMessage()));
+                                            return null;
+                                        });
+                    } else if (commandEndTxnOnPartition.getTxnAction().equals(PulsarApi.TxnAction.ABORT)) {
+                        markerController.publishAbortMarker()
```
[the remark this answers is not in the document]
Comment (zymap) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java]: Sorry. It's my mistake.

**Pull request closed without merging**